packages/instant/src/redux/async_data.ts

```diff
@@ -33,4 +34,40 @@ export const asyncData = {
             store.dispatch(actions.setAvailableAssets([]));
         }
     },
+    fetchAccountInfoAndDispatchToStore: async (store: Store) => {
+        const { providerState } = store.getState();
+        const web3Wrapper = providerState.web3Wrapper;
+        if (providerState.account.state !== AccountState.Loading) {
+            store.dispatch(actions.setAccountStateLoading());
+        }
+        let availableAddresses: string[];
+        try {
+            availableAddresses = await web3Wrapper.getAvailableAddressesAsync();
+        } catch (e) {
+            store.dispatch(actions.setAccountStateError());
+            return;
+        }
+        if (!_.isEmpty(availableAddresses)) {
```

Comment:
super nit (not necessary to change) but I'd prefer for this to follow the `return` pattern above, i.e.

```
if (_.isEmpty(availableAddresses)) {
   store.dispatch(actions.setAccountStateLocked());
   return;
}

... setAccountStateReady code ..
```